### PR TITLE
Add support for variables

### PR DESCRIPTION
**Pull request by TucksonDev**
Attempt to add support for variables within the docs. This PR presents the basic structure and a use case: node images

Current behavior is:
- Any text, code or URL (in link tags) enclosed in @ characters can be replaced by the correspondent variable (e.g. @latestNitroVersion@)
- Variables are kept in `website/static/globalVars.js`

There are some limitations for this support:
- We can use other pattern, but if we want to support injecting variables in backticks enclosed texts, we can't use the following characters: [, {, (, |, &, = (probably some more). These characters are handled differently by docusaurus when converting those texts to "code" tags, to support syntax highlighting.
- Variables inside single backticks are not detected, but those can be replaced by < code> HTML tags (the behavior is the same)

### CODE REVIEW

**Review by mahsamoosavi — APPROVED** [review]
This will make our lives much easier! 🧚🏻‍♀️

File: website/src/remark/variable-injector.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+//    @ FOO @ => /\@\s*([a-zA-Z0-9_-]+)\s*\@/g
+//    :: FOO :: => /\:\:\s*([a-zA-Z0-9_-]+)\s*\:\:/g
+
+const visit = require('unist-util-visit');
+
+const plugin = (options) => {
+  const transformer = async (ast) => {
+    visit(ast, ['text', 'code', 'link'], (node) => {
+      // Replace all occurrences of [[ varName ]] with the value of varName
+      let value;
```
Comment (DZGoldman) [inline website/src/remark/variable-injector.js:21]: could use iffy syntax here (no necessary tho)

File: website/src/remark/variable-injector.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          value = node.url;
+          break;
+        
+        case "text":
+        case "code":
+          value = node.value;
+          break;
+      }
+
+      value = value.replace(/\@\s*([a-zA-Z0-9_-]+)\s*\@/g, (match, varName) => {
```
Comment (DZGoldman) [inline website/src/remark/variable-injector.js:33]: comment explaining in words? (I think just "stuff between two "@"s)?

**Review by DZGoldman — CHANGES_REQUESTED** [review]
pumped

**Comment by TucksonDev** [comment]
@DZGoldman , I forgot to revisit that file after finishing with the rest of changes. I've added more clarification and removed outdated comments :)

**Review by symbolpunk — APPROVED** [review]

**Review by DZGoldman — APPROVED** [review]
lgtm

**Pull request merged**